src/gnn_tracking/postprocessing/fastrescanner.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+from sklearn.cluster._dbscan_inner import dbscan_inner
+from sklearn.cluster import DBSCAN
+
+class DBSCANFastRescan:
+
+    def __init__(self, X, max_eps=1, n_jobs=None):
+        """Class to perform DBSCAN clustering with fast rescanning.
+
+        Args:
+            X: Data to cluster
+            max_eps: Maximum epsilon to use during rescanning. Set to as low
+                as possible to save time.
+            n_jobs : int, default=None
+                The number of parallel jobs to run.
+        """
+        self.X = X
+        self.max_eps = max_eps
+        self.n_jobs=n_jobs
+        self._reset_graph(max_eps)
+
+    def _reset_graph(self, max_eps):
+        """
+        Set and store the radius_neighbors graph to use for clustering.
+        """
+        neighbors_model = NearestNeighbors(radius=max_eps)
+        neighbors_model.fit(self.X)
+        distances, indexes = neighbors_model.radius_neighbors(self.X, radius=max_eps, return_distance=True, n_jobs=self.n_jobs)
+
+        lengths = [len(arr) for arr in indexes]
+        source_indexes = np.repeat(np.arange(len(indexes)), lengths)
+        target_indexes = np.concatenate(indexes)
+
+        self.distances = np.concatenate(distances)
```

Comment:
all attributes should be declared in `__init__`. Also mark as private (start with `_`)